### PR TITLE
sdk generation pipeline support java dpg

### PR DESCRIPTION
**Pull request by zzvswxy**
Signed-off-by: Zhou Zheng <zhouzheng@microsoft.com>

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

Make sdk generation pipeline script support java dpg

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/mgmt/automation/sdk_init.sh [hunk 1]
```diff
@@ -1,9 +1,8 @@
 #!/bin/sh
 
-sudo apt-get install -y --upgrade python3-pip python3-setuptools
-pip3 install --upgrade wheel
-pip3 install --upgrade PyYAML requests
+apt-get install python3.8
+apt-get install -y --upgrade python3-pip python3-setuptools
```
Comment (weidongxu-microsoft) [inline eng/mgmt/automation/sdk_init.sh:4]: Is this works same on sdkautomation?

**Review by weidongxu-microsoft — APPROVED** [review]

File: eng/mgmt/automation/generate_data.py [hunk 2]
```diff
@@ -23,35 +23,9 @@
 YAML_BLOCK_REGEX = r'```\s?(?:yaml|YAML).*?\n(.*?)```'
 
 
-def sdk_automation(config: dict) -> List[dict]:
-    # priority:
-    # 1. autorestConfig from input
-    # 2. swagger/README.md in sdk repository that matches readme from input
-
+def get_sdk_readme_abspath(config: dict, readme_file_path: str) -> Optional[str]:
```
Comment (weidongxu-microsoft) [inline eng/mgmt/automation/generate_data.py:26]: this is actually `get_or_update_sdk_readme`?

File: eng/mgmt/automation/sdk_init.sh (hunk 1 above)
Comment (zzvswxy) [reply eng/mgmt/automation/sdk_init.sh:4]: sdk_init.sh only used in sdk generation pipeline docker. Docker run script in root user, so we don't need sudo cmd. Script has tested in pipeline docker and local docker.

File: eng/mgmt/automation/generate_data.py (hunk 2 above)
Comment (zzvswxy) [reply eng/mgmt/automation/generate_data.py]: you are right, the func may get the readme file or update it, fixed

**Pull request merged**